src/ui/public/index_patterns/__tests__/_index_pattern.js

```diff
@@ -405,12 +404,48 @@ describe('index pattern', function () {
 
   describe('#isWildcard()', function () {
     it('returns true if id has an *', function () {
-      indexPattern.id = 'foo*';
-      expect(indexPattern.isWildcard()).to.be(true);
+      return create('foo*').then(function (pattern) {
+        expect(pattern.isWildcard()).to.be(true);
+      });
     });
     it('returns false if id has no *', function () {
-      indexPattern.id = 'foo';
-      expect(indexPattern.isWildcard()).to.be(false);
+      return create('foo').then(function (pattern) {
+        expect(pattern.isWildcard()).to.be(false);
+      });
+    });
+  });
+
+  describe('#create()', function () {
+    require('testUtils/noDigestPromises').activateForSuite();
+
+    it('should require an id', function () {
+      return create()
+      .then(indexPattern => indexPattern.create())
+      .then(
+        () => {
+          throw new Error('expected create to fail');
+        },
+        (err) => {
+          expect(err.message).to.match(/\bid\b/i);
+        }
```

Comment:
Ah, right, it's a promise rejection not an exception. Brain not working. An assertion library that supported promises sure would be nice.